drivers/backup/portworx/portworx.go

```diff
@@ -1535,6 +1617,166 @@ func (p *portworx) DeleteBackupSchedulePolicy(orgID string, policyList []string)
 	return nil
 }
 
+// CreateRuleForBackupWithMultipleApplications creates backup rule for multiple application in one rule
+func (p *portworx) CreateRuleForBackupWithMultipleApplications(orgID string, appList []string, ctx context.Context, appParameters ...map[string]backup.AppRule) (string, string, error) {
+	var (
+		preUid             string
+		preRuleName        string
+		postRuleName       string
+		postUid            string
+		preActionValue     []string
+		preContainer       []string
+		postActionValue    []string
+		postContainer      []string
+		postBackground     []bool
+		postRunInSinglePod []bool
+		preBackground      []bool
+		preRunInSinglePod  []bool
+		preRulesInfo       api.RulesInfo
+		postRulesInfo      api.RulesInfo
+		prePodSelector     []map[string]string
+		postPodSelector    []map[string]string
+	)
+	if len(appParameters) == 0 {
+		appParameters = append(appParameters, AppRuleMaster)
+	}
+
+	for i := 0; i < len(appList); i++ {
+		if reflect.DeepEqual(appParameters[0][appList[i]].PreRule, backup.PreRule{}) {
```

Comment:
I see that we are using `appParameters[0][appList[i]]` at many places. Can we hold this in a variable and use it across?